dot/sync/disjoint_block_set.go

```diff
@@ -26,7 +27,7 @@ var (
 // DisjointBlockSet represents a set of incomplete blocks, or blocks
 // with an unknown parent. it is implemented by *disjointBlockSet
 type DisjointBlockSet interface {
-	run(done <-chan struct{})
+	run(finalisedCh <-chan *types.FinalisationInfo, done <-chan struct{})
```

Comment:
`done` is usually reserved to signal that something has completed.
```suggestion
	run(finalisedCh <-chan *types.FinalisationInfo, stop <-chan struct{})
```